### PR TITLE
Enabling BH L1 data cache

### PR DESCRIPTION
**Pull request by abhullar-tt**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/10865

### Problem description
L1 cache was default disabled on BH

### What's changed
- Enabling L1 cache and adding api to invalidate cache. Added invalidation to apis that poll a L1 locations
- Added debug env var `TT_METAL_DISABLE_L1_DATA_CACHE_RISCVS` that can be set to one or more of <BR,NC,TR,ER> to disable the cache

### Checklist
- [x] [Post commit CI passes](https://github.com/tenstorrent/tt-metal/actions/runs/10565994280)
- [x] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/runs/10566009712)

### CODE REVIEW

**Comment by ubcheema** [comment]
There is no separate instruction/data cache?

**Comment by abhullar-tt** [comment]
> There is no separate instruction/data cache?

this change is just for l1 data cache. updated the pr description

File: tt_metal/impl/dispatch/kernels/cq_dispatch.cpp [hunk 1]
```diff
@@ -875,13 +878,13 @@ static inline bool process_cmd_d(uint32_t &cmd_ptr) {
     switch (cmd->base.cmd_id) {
         case CQ_DISPATCH_CMD_WRITE_LINEAR:
             DEBUG_STATUS("DWB");
-            DPRINT << "cmd_write\n";
+            // DPRINT << "cmd_write\n";
```
[the remark this answers is not in the document]
Comment (abhullar-tt) [reply tt_metal/impl/dispatch/kernels/cq_dispatch.cpp:881]: ah yeah, I think this was mistakenly checked in

**Review by ubcheema — APPROVED** [review]

**Review by tt-aho — APPROVED** [review]
Do we need to invalidate cache in cq_prefetch `process_prefetch_h_wait_cmd` fn or when reading `prefetch_q_rd_ptr` in this loop?
```
            while ((fetch_size = *prefetch_q_rd_ptr) == 0) {
                IDLE_ERISC_HEARTBEAT_AND_RETURN(heartbeat);
            }
```

**Review by tt-dma — APPROVED** [review]

**Review by tt-asaigal — APPROVED** [review]

**Comment by abhullar-tt** [comment]
> Do we need to invalidate cache in cq_prefetch `process_prefetch_h_wait_cmd` fn or when reading `prefetch_q_rd_ptr` in this loop?
> 
> ```
>             while ((fetch_size = *prefetch_q_rd_ptr) == 0) {
>                 IDLE_ERISC_HEARTBEAT_AND_RETURN(heartbeat);
>             }
> ```

good point, essentially any address written by noc that a reader may have previously read needs to be invalidated

**Comment by abhullar-tt** [comment]
> process_prefetch_h_wait_cmd

added invalidate to a couple spots in cq_prefetch

**Pull request merged**